spring-boot-project/spring-boot-devtools/src/test/java/org/springframework/boot/devtools/livereload/LiveReloadServerTests.java

```diff
@@ -88,19 +89,14 @@ void servesLivereloadJs() throws Exception {
 	void triggerReload() throws Exception {
 		LiveReloadWebSocketHandler handler = connect();
 		this.server.triggerReload();
-		Thread.sleep(200);
-		this.server.stop();
-		assertThat(handler.getMessages().get(0)).contains("http://livereload.com/protocols/official-7");
-		assertThat(handler.getMessages().get(1)).contains("command\":\"reload\"");
+		await().atMost(Duration.ofSeconds(1)).until(handler::getMessages, msgs -> msgs.get(0).contains("http://livereload.com/protocols/official-7") && msgs.get(1).contains("command\":\"reload\""));
 	}
 
 	@Test
 	void pingPong() throws Exception {
 		LiveReloadWebSocketHandler handler = connect();
 		handler.sendMessage(new PingMessage());
-		Thread.sleep(200);
-		assertThat(handler.getPongCount()).isEqualTo(1);
-		this.server.stop();
```

Comment:
I think we need to keep this call to `stop()`.

Reply:
The same situation as [above](https://github.com/spring-projects/spring-boot/pull/21988#discussion_r441769817).